chat/src/main/java/com/github/twitch4j/chat/TwitchChatConnectionPool.java

```diff
@@ -0,0 +1,163 @@
+package com.github.twitch4j.chat;
+
+import com.github.philippheuer.credentialmanager.domain.OAuth2Credential;
+import com.github.twitch4j.common.pool.TwitchModuleConnectionPool;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.experimental.SuperBuilder;
+import org.apache.commons.lang3.RandomStringUtils;
+
+import java.util.function.Supplier;
+
+/**
+ * A pool for {@link TwitchChat} connections.
+ * <p>
+ * This pool is easiest to use for:
+ * <ul>
+ *     <li>Reading from many channels without an account</li>
+ *     <li>Reading from many channels with an account</li>
+ *     <li>Reading from many channels using valid proxies</li>
+ *     <li>Reading and sending messages/whispers with only one account</li>
+ * </ul>
+ * Other applications are possible, but harder to configure. See below for a list of warnings.
+ * <p>
+ * Warning: If a custom {@link java.util.concurrent.ScheduledThreadPoolExecutor} is specified,
+ * its corePoolSize must be large enough for the threads required by connections made by this class.
```

Comment:
Should make `TwitchChat` more thread-efficient (but this may be a task for another PR)